inductiva/resources/machines_base.py

```diff
@@ -55,15 +56,15 @@ def __init__(self,
         """
 
         provider = ProviderType(provider.upper())
+        self.provider = provider.value
 
-        if machine_type not in list_available_machines(provider.value.lower()):
-            raise ValueError(f"Machine type not supported in {provider}")
+        if machine_type not in list_available_machines(self.provider.lower()):
```

Comment:
`list_available_machines` should receive either a `str` or a `ProviderType`.

```python
def list_available_machines(provider: str|ProviderType):
    provider = ProviderType(provider)   # if ProviderType is case insensitive, this will work
    ...
```

```suggestion
        if machine_type not in list_available_machines(self.provider):
```